### PR TITLE
Remove dangerouslySetInnerHTML from CodeSnippet

### LINKED ISSUE #34: Display test context code without rendering HTML
Is it possible for the test context content to display without rendering as HTML? I noticed that in [`code-snippet.jsx`](https://github.com/adamgruber/mochawesome-report-generator/blob/master/src/components/test/code-snippet.jsx#L59), the `CodeSnippet` component uses `dangerouslySetInnerHTML`.

Is there a reason for this? I am displaying HTML code in the test context and I believe it shouldn't be rendering (since it's a `CodeSnippet` after all). Plus, this might open up XSS risks (I'm personally using this to generate dynamic test reports).

I would be happy to create a PR to fix this if neeeded.

### PR DESCRIPTION
**Pull request by irvinlim**
Fixes #34.

### CODE REVIEW

File: test/spec/components/test/code-snippet.test.jsx [hunk 1]
```diff
@@ -82,4 +82,12 @@ describe('<CodeSnippet />', () => {
     });
     expect(CodeSnippet.prototype.shouldComponentUpdate.calledOnce).to.equal(true);
   });
+
+  it('does not render HTML tags', () => {
```
Comment (adamgruber) [inline test/spec/components/test/code-snippet.test.jsx]: Maybe reword this a little. `it('does not render HTML as markup')`

File: src/components/test/code-snippet.jsx [hunk 2]
```diff
@@ -43,20 +43,22 @@ class CodeSnippet extends Component {
...
-      codeHtml = `${expected}&nbsp;&nbsp;${actual}\n\n${code}`;
-    }
-
     const cxName = cx(className, lang, { hljs: !highlight });
 
     return !!code && (
       <pre className={ cxName } ref={ node => (this.node = node) }>
-        <code dangerouslySetInnerHTML={ { __html: codeHtml } } />
+        <code>
+          { lang === 'diff' &&
```
Comment (adamgruber) [inline src/components/test/code-snippet.jsx]: I think this can be cleaned up a bit with some CSS.
```jsx
<code>
  { lang === 'diff' && 
    <div className={ cx('code-diff-header') }>
      <span className={ cx('code-diff-expected') }>+ expected</span>
      <span className={ cx('code-diff-actual') }>- actual</span>
    </div>
  }
  { code }
</code>
```

We can use styles to add the space between `expected` and `actual` and then to add the space below the diff header.

File: test/spec/components/test/code-snippet.test.jsx (hunk 1 above)
Comment (adamgruber) [inline test/spec/components/test/code-snippet.test.jsx]: nit: Let's move this test up next to the other render tests

**Comment by irvinlim** [comment]
@adamgruber Fixed!
